### PR TITLE
fix: convert AF1 fragment containers; guard optional property removal

### PR DESCRIPTION
**Pull request by im-shiv**
Two root causes prevented all fragment containers from converting:

1. ContainerRewriterRule: getProperty(GUIDE_NODE_CLASS/GUIDE_CSS).remove() without hasProperty() check. Fragment containers lack these optional AF1 UI properties. PathNotFoundException was swallowed by the catch block; v2Path was never set, stalling RootPanelRule and CommonGuideComponentsRewriterRule for the fragment.

2. AdaptiveFormUtils.deleteFormNodes: rename step only handled guideContainer and guideContainerWrapper. Fragment containers (guideFragmentContainer) fell through to recursive branch which hit the new AF2 container node (has sling:resourceType), causing checkIfNodeCanBeDeleted to return false — temp container never renamed.

Additional fixes in AdaptiveFormUtils:
- handleLayoutNode: add hasProperty guard for GUIDE_NODE_CLASS
- handleLayoutNode else branch: set CORE_PANEL_RESOURCE_TYPE + FIELD_TYPE on plain panels (was carrying over AF1 sling:resourceType)
- populateResourceTypeForLayout: fix VERTICAL_TABS <-> TABS_ON_TOP argument swap

Defensive hasProperty guards in PanelRule.handleFragmentPanel and TableRule (getTableContainer, processTableNode rows/cells, copyTableNodeChildren) — same unsafe pattern that caused the ContainerRule failure.

AdaptiveFormConstants: add 'toolbar' to PANEL_NODES_TO_IGNORE to prevent toolbar nodes from being duplicated during panel copy.

### CODE REVIEW

File: core/src/main/java/com/adobe/aem/core/utils/AdaptiveFormConstants.java [hunk 1]
```diff
@@ -149,6 +149,6 @@ public class AdaptiveFormConstants {
     public static final String CORE_SUBMIT_ACTION_RESOURCE_TYPE = COMPONENT_PATH_PREFIX + "actions/submit";
     public static final String CORE_RESET_ACTION_RESOURCE_TYPE = COMPONENT_PATH_PREFIX + "actions/reset";
     public final static List<String> GRID_LAYOUTS = Arrays.asList("fd/af/layouts/gridFluidLayout", "fd/af/layouts/gridFluidLayout2", "fd/af/layouts/defaultGuideLayout");
-    public final static List<String> PANEL_NODES_TO_IGNORE = Arrays.asList("layout", "items");
+    public final static List<String> PANEL_NODES_TO_IGNORE = Arrays.asList("layout", "items", "toolbar");
```
Comment (iamsudhanshu) [inline core/src/main/java/com/adobe/aem/core/utils/AdaptiveFormConstants.java:152]: Toolbar is actually a panel container with pre-defined buttons. Any reason we cannot migrate this.

Comment (im-shiv) [reply core/src/main/java/com/adobe/aem/core/utils/AdaptiveFormConstants.java:152]: In AF1, the toolbar holds buttons like Submit, Reset, Next, and Previous. Submit and Reset have CC equivalents (actions/submit, actions/reset). However, Next and Previous are wizard-navigation constructs that don't exist as standalone components in CC — the wizard layout in CC handles navigation internally.

Migrating the toolbar would mean copying Submit/Reset into the new container while silently dropping Next/Previous, which would leave the converted form in an inconsistent state (buttons referencing navigation behaviour that no longer exists). Ignoring the toolbar entirely is safer — the converted form's submit/reset buttons can be re-added explicitly, and wizard navigation is handled by the CC wizard layout component itself.

If you'd like, we can add targeted migration for just the Submit and Reset buttons from the toolbar in a follow-up — but that would require a separate rule that explicitly handles fd/af/components/toolbar and only copies children with known CC equivalents.

**Comment by iamsudhanshu** [comment]
Review comments by Claude:
 PR #5 Review — fix: convert AF1 fragment containers; guard optional property removal

  **Overview**

  This PR fixes a cascading failure that prevented all AF1 fragment containers from converting to AF2 core components. The root cause was getProperty().remove() calls on optional AF1 UI properties without a hasProperty() guard — when these
  properties were absent, a PathNotFoundException was swallowed by the catch block, leaving v2Path unset and stalling the downstream RootPanelRule and CommonGuideComponentsRewriterRule. Secondary fixes correct a deleteFormNodes rename gap, a
  significant argument-swap bug in populateResourceTypeForLayout, and a missing resource-type assignment on plain panels.
  
  ---
  **Correctness**
  
  hasProperty() guards (ContainerRule, PanelRule, TableRule)
  All correct. The guards follow the same pattern already used elsewhere in the codebase (e.g., isRepeatablePanel). These are straightforward defensive fixes.

  deleteFormNodes — fragment container rename (AdaptiveFormUtils.java:53)
  Correct fix. GUIDE_FRAGMENT_CONTAINER = "guideFragmentContainer" (constant at AdaptiveFormConstants.java:25) matches the JCR node name, so parent.getName().equals(GUIDE_FRAGMENT_CONTAINER) correctly short-circuits the recursive branch and
  triggers the temp-container rename.

  populateResourceTypeForLayout argument swap (AdaptiveFormUtils.java:91-94)
  This is the most impactful fix. The call site in AdaptiveFormGuidePanelRewriterRule.activate() (line 58–59) passes arguments in order (WIZARD, TABS_ON_TOP, VERTICAL_TABS, ACCORDION). The old mapping had the 2nd and 3rd parameters inverted —
  every form using vertical tabs would silently render with the tabs-on-top layout and vice versa. The fix is correct.

  handleLayoutNode else branch — missing resource type (AdaptiveFormUtils.java:79-80)
  Correct. createNodeAndCopyProperties copies all non-protected JCR properties including the AF1 sling:resourceType. Without explicitly overwriting it, plain panels would carry the AF1 resource type into the AF2 structure. Setting
  CORE_PANEL_RESOURCE_TYPE + FIELD_TYPE here closes that gap.

  toolbar added to PANEL_NODES_TO_IGNORE (AdaptiveFormConstants.java:152)
  Correct. This prevents toolbar nodes (which carry sling:resourceType) from blocking checkIfNodeCanBeDeleted and from being duplicated during panel copy.

  ---
  **Issues and Risks**
  
  **1. Style inconsistency — braceless if statements**
  The new guards use single-line if without braces:
  if (container.hasProperty(GUIDE_NODE_CLASS)) container.getProperty(GUIDE_NODE_CLASS).remove();
  The rest of the codebase uses braces consistently. This is a minor style concern but a common source of future bugs (Apples SSL goto-fail style). Prefer:
  if (container.hasProperty(GUIDE_NODE_CLASS)) {
      container.getProperty(GUIDE_NODE_CLASS).remove();
  }   
  
  **2. populateResourceTypeForLayout signature remains fragile**
  The argument swap was possible because the parameter names (coreTabsOnTopResourceType, coreVerticalTabsResourceType) are easy to confuse. Now that it's been fixed once, this could easily regress. Consider adding inline comments or reordering
  the parameters to match the LAYOUT_RESOURCE_TYPE_MAPPINGS.put() order, or at minimum documenting the expected call order.

  **3. FIELD_TYPE on fragment containers is still "form"**
  In ContainerRewriterRule.processComponent() (line 71): container.setProperty(FIELD_TYPE, FIELD_TYPE_FORM) is set unconditionally before the branch that sets CORE_FRAGMENT_CONTAINER_RESOURCE_TYPE. Fragment containers then get fd:type = 
  "fragment" but fieldType remains "form". This is a pre-existing issue, not introduced by this PR — but it's worth tracking.

  **4. copyTableNodeChildren removes property from source node**
  JcrUtil.copy(node, panelNode, node.getName());
  if (node.hasProperty(SLING_RESOURCE_TYPE_PROPERTY)) node.getProperty(SLING_RESOURCE_TYPE_PROPERTY).remove();
  The property is removed from the source (AF1 node) after copying to the destination. This is the existing pattern, not a regression introduced here. The hasProperty guard just makes it safe. No new risk.

  **5. No test coverage for fragment container path**
  The only test class (AdaptiveFormTreeRewriteRuleTest) is @Disabled. These bugs clearly existed undetected because no integration tests exercise the fragment container conversion path. Without new tests, this fix could regress silently.

  ---
  **Suggestions**
  
  - Add braces to the new single-line if guards (all 7 occurrences).
  - Add a regression test for fragment container conversion — at minimum a @Test that asserts guideFragmentContainer → fragmentcontainer resource type and that guideNodeClass/guideCss are absent on the output.
  - Consider a comment or rename near populateResourceTypeForLayout's parameters to make the expected argument order obvious.
  - Enable or rewrite AdaptiveFormTreeRewriteRuleTest — having no active tests is the primary ongoing risk for this whole module.

File: core/src/main/java/com/adobe/aem/core/utils/AdaptiveFormConstants.java (hunk 1 above)
Comment (iamsudhanshu) [reply core/src/main/java/com/adobe/aem/core/utils/AdaptiveFormConstants.java:152]: The current approach removes - toolbar, back, next, submit, save, reset - while save, reset, submit can be resolved keeping toolbar as a panel. Ideally - instead of silently removing - log should indicate removal of Previous/Next - and other available components should be migrated.
**Note:** FORMS-16339 tracks - Creation of rule for conversion of toolbar into panel with next and previous buttons with rules.
